src/Monitor.ts

```diff
@@ -0,0 +1,67 @@
+import axios from 'axios';
+import { Observable, Subscription } from 'rxjs';
+import { filter } from 'rxjs/operators';
+import * as shell from 'shelljs';
+import { NetworkType, MonitorInterface, MonitorConfig, TaskInterface, ActionPOST, ActionScript } from './types';
+import getTask from './getTask';
+
+export default class Monitor implements MonitorInterface {
+  public readonly name: string;
+  public readonly config: MonitorConfig;
+  public readonly task: TaskInterface;
+  public readonly rawOutput$: Observable<any>;
+  public readonly output$: Observable<any>;
+
+  constructor(name: string, network: NetworkType, config: MonitorConfig) {
+    this.name = name;
+    this.config = config;
+
+    this.task = getTask(network, config.task);
+    if (!this.task) {
+      throw Error(`${name}:${config.task} not found`);
+    }
+
+    // create raw output$
+    this.rawOutput$ = this.task.call(config.arguments);
+
+    // create filtered output$
+    this.output$ = this.rawOutput$.pipe(
+      filter(() => {
+        if (this.config.conditions) {
+          // TODO: check conditions
+          return true;
+        }
+        return true;
+      })
+    );
+  }
+
+  /// action post
+  post(action: ActionPOST, data: any) {
+    const { method, url, headers } = action;
+    axios.request({ url, method, headers, data });
+  }
+
+  /// action script
+  script(action: ActionScript, data: any) {
+    const dataStr = JSON.stringify(data);
+    const command = `sh ${action.path} '${dataStr}'`;
```

Comment:
script does not need to be shell script, it can be any executable file. e.g. js file with valid shebang
also pipe data via stdin instead argument. the data can be any format and we don't want to deal with all the edge cases with shell.

Reply:
yes going to updated that